specs/interop.md

```diff
@@ -0,0 +1,396 @@
+    require(L1Block.isInInteropSet(_msg.chainid));
+
+    // rest of implementation
+}
+```
+
+By including the hash in the calldata, it makes static analysis much easier for block builders.
+Without the check onchain, the block builder would need to hash the `CrossChainMessage` to build the
+`eth_getLogs` query. With the hash onchain, the block builder can directly trust the hash.
+
+### L1Block
+
+The `L1Block` contract is updated to include the set of allowed chains. The L1 Atrributes transaction
+sets the set of allowed chains. The `L1Block` contract MUST provide a public getter to check is a particular
+chain is in the dependency set.
+
+The `setL1BlockValuesInterop()` function MUST be called on every block after the interop upgrade block.
+The interop upgrade block itself MUST include a call to `setL1BlockValuesEcotone`.
+
+## SystemConfig
+
+The `SystemConfig` is updated to manage the dependency set. The chain operator can add or remove chains from the dependency set
+through the `SystemConfig`. A new `ConfigUpdate` event `UpdateType` enum is added that corresponds to a change in the dependency set.
+
+The `SystemConfig` MUST enforce that the maximum size of the dependency set is `type(uint8).max` or 255.
+
+### `DEPENDENCY_SET` UpdateType
+
+When a `ConfigUpdate` event is emitted where the `UpdateType` is `DEPENDENCY_SET`, the L2 network will update its dependency set.
+The chain operator SHOULD be able to add or remove chains from the dependency set.
+
+## L1Attributes
+
+The L1 Atrributes transaction is updated to include the dependency set. Since the dependency set is dynamically sized,
+a `uint8` "interopSetSize" parameter prefixes tightly packed `uint256` values that represent each chain id.
+
+| Input arg         | Type                    | Calldata bytes          | Segment |
+| ----------------- | ----------------------- | ----------------------- | --------|
+| {0x760ee04d}      | bytes4                  | 0-3                     | n/a     |
+| baseFeeScalar     | uint32                  | 4-7                     | 1       |
+| blobBaseFeeScalar | uint32                  | 8-11                    |         |
+| sequenceNumber    | uint64                  | 12-19                   |         |
+| l1BlockTimestamp  | uint64                  | 20-27                   |         |
+| l1BlockNumber     | uint64                  | 28-35                   |         |
+| basefee           | uint256                 | 36-67                   | 2       |
+| blobBaseFee       | uint256                 | 68-99                   | 3       |
+| l1BlockHash       | bytes32                 | 100-131                 | 4       |
+| batcherHash       | bytes32                 | 132-163                 | 5       |
+| interopSetSize    | uint8                   | 164-165                 | 6       |
+| chainIds          | [interopSetSize]uint256 | 165-(32*interopSetSize) | 6+      |
+
+## Fault Proof
+
+Pseudocode for the `op-program`
```

Comment:
I probably just need the `TODO: more description` part to give me the right context here, but this code doesn't map to my mental model of op-program at all so I can't make any sense of it. :)